src/check/canonicalize/CIR.zig

```diff
@@ -887,14 +892,37 @@ pub const Expr = union(enum) {
                 elem_var_node.appendString(gpa, elem_var_str);
                 list_node.appendNode(gpa, &elem_var_node);
 
-                // TODO print list elems
-                // implement proper span access when collection is available
+                // Add list elements
                 var elems_node = sexpr.Expr.init(gpa, "elems");
-                elems_node.appendString(gpa, "TODO each element");
+                for (ir.store.sliceExpr(l.elems)) |elem_idx| {
+                    var elem_node = ir.store.getExpr(elem_idx).toSExpr(ir, env);
+                    elems_node.appendNode(gpa, &elem_node);
+                }
                 list_node.appendNode(gpa, &elems_node);
 
                 return list_node;
             },
+            .tuple => |t| {
+                var tuple_node = sexpr.Expr.init(gpa, "e_tuple");
+                tuple_node.appendRegionInfo(gpa, ir.calcRegionInfo(t.region));
+
+                // Add tuple_var
+                var tuple_var_node = sexpr.Expr.init(gpa, "tuple_var");
+                const tuple_var_str = t.tuple_var.allocPrint(gpa);
+                defer gpa.free(tuple_var_str);
+                tuple_var_node.appendString(gpa, tuple_var_str);
+                tuple_node.appendNode(gpa, &tuple_var_node);
```

Comment:
I updates most of these in [my other PR](https://github.com/roc-lang/roc/pull/7848/commits/cb584f96cb517ee0ed747493dd20b93f1d323b52#diff-e24da2a73205d9269f8d13301fa364218cf0aa44815ac6039e9e637dd924697d), but can we write type variables as numbers instead of strings?

```zig
var tuple_var_node = sexpr.Expr.init(gpa, "tuple_var");
tuple_var_node.appendUnsignedInt(gpa, t.tuple_var);
tuple_node.appendNode(gpa, &tuple_var_node);
```